fe/fe-core/src/main/java/org/apache/doris/mtmv/MTMVCache.java

```diff
@@ -68,13 +77,28 @@ public static MTMVCache from(MTMV mtmv, ConnectContext connectContext) {
         }
         Plan mvRewrittenPlan =
                 planner.plan(unboundMvPlan, PhysicalProperties.ANY, ExplainLevel.REWRITTEN_PLAN);
-        Plan mvPlan = mvRewrittenPlan instanceof LogicalResultSink
-                ? (Plan) ((LogicalResultSink) mvRewrittenPlan).child() : mvRewrittenPlan;
+        // remove top sort and result sink node above materialized view plan
+        if (mvRewrittenPlan instanceof LogicalResultSink) {
+            // replace result sink to table sink for sort eliminate
+            mvRewrittenPlan = new LogicalOlapTableSink<>(new Database(), mtmv, mtmv.getBaseSchema(),
+                    mtmv.getPartitionIds(),
+                    mvRewrittenPlan.getOutput().stream().map(NamedExpression.class::cast).collect(Collectors.toList()),
+                    mtmv.isPartitionedTable(), DMLCommandType.NONE, mvRewrittenPlan.child(0));
+            // eliminate sort under sink because sort is useless for materialized view
+            planner.getCascadesContext().setRewritePlan(mvRewrittenPlan);
+            Rewriter rewriter = Rewriter.getCteChildrenRewriter(planner.getCascadesContext(),
+                    Lists.newArrayList(AbstractBatchJobExecutor.topDown(new EliminateSortUnderSink())));
+            rewriter.execute();
+        }
+        if (planner.getCascadesContext().getRewritePlan() instanceof Sink) {
+            // remove the sink
+            mvRewrittenPlan = planner.getCascadesContext().getRewritePlan().child(0);
+        }
```

Comment:
should notice, result sink is not always the top plan node after plan. so u should replace result sink with table sink before plan. and then when u do plan, the top sort node will be removed

Reply:
Have fixed it